feat: migrate e2e tests

temaniarpit27 opened the pull request:
https://github.com/0xPolygon/cdk/issues/327
https://github.com/0xPolygon/cdk/issues/328

temaniarpit27 committed:
add e2e

temaniarpit27 committed:
fix

temaniarpit27 committed:
fix

temaniarpit27 committed:
fix

temaniarpit27 committed:
fix

temaniarpit27 committed:
fix

temaniarpit27 committed:
fix

temaniarpit27 committed:
fix

temaniarpit27 committed:
fix

temaniarpit27 committed:
fix

temaniarpit27 committed:
fix

temaniarpit27 committed:
change hash

temaniarpit27 committed:
fix

temaniarpit27 committed:
debugging

temaniarpit27 committed:
version change

temaniarpit27 committed:
version change

temaniarpit27 committed:
add l1infotree

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fatalf to errorf

temaniarpit27 committed:
fix args

temaniarpit27 committed:
Merge branch 'arpit/aggkit-409' of github.com:0xPolygon/cdk into arpit/325-2

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
update go mod

temaniarpit27 committed:
update go mod

temaniarpit27 committed:
review comments fix

temaniarpit27 committed:
review comments fix

temaniarpit27 committed:
review comments fix

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
revert back

temaniarpit27 committed:
Merge branch 'arpit/325-2' of github.com:0xPolygon/cdk into arpit/325-2

temaniarpit27 committed:
fatalf to panicf

temaniarpit27 committed:
merge base

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
fix ci

temaniarpit27 committed:
merge develop

Stefan-Ethernal commented on test/run-local-e2e.sh:
```diff
@@ -0,0 +1,96 @@
+
+pushd "$E2E_FOLDER" > /dev/null
+
+# Setup environment
+log_info "Setting up e2e environment..."
+set -a
+source ./tests/.env
+set +a
+
+export BATS_LIB_PATH="$PWD/core/helpers/lib"
+export PROJECT_ROOT="$PWD"
+export ENCLAVE="$ENCLAVE"
+export DISABLE_L2_FUND="true"
+
+log_info "Running BATS E2E tests..."
+bats ./tests/cdk
+
+popd > /dev/null
+log_info "E2E tests executed. Logs saved to $LOG_FILE"
```
Can we introduce parameter that would allow us to only run the Kurtosis environment, without running the e2e tests? I wanted to point this out for the aggkit as well, but I've forgotten and the PR got merged in a meantime. WDYT?

temaniarpit27 replied:
ya i was also thinking the same

temaniarpit27 replied:
it will be fixed in next PR

Stefan-Ethernal commented on .github/workflows/test-e2e.yml:
```diff
@@ -30,79 +30,148 @@
+        }
+
+  check-fork12-rollup-e2e-result:
+    name: Check E2E results
+    needs:
+      - test-fork12-rollup-e2e
+    if: always()
+    runs-on: ubuntu-latest
+    steps:
+      - run: exit 1
+        if: ${{ contains(fromJSON('["failure", "cancelled"]'), needs.test-fork12-rollup-e2e.result) }}
```
@temaniarpit27 do you understand this warning and why it is fired? I haven't investigated about it, but at first glance I don't understand.

temaniarpit27 replied:
https://github.com/0xPolygon/cdk/pull/316/files#diff-976614657a9537dd6476a2977a4a11c0ac912e262006a4ca2d4fe0b4d8cf0becR9

basically some permissions are required for this in cdk

Stefan-Ethernal reviewed (commented):
Update the `Makefile`, as the commands are outdated (https://github.com/0xPolygon/cdk/blob/94718685ad8a3df9d2975649b1c761aa53ed316b/test/Makefile#L7-L27)

temaniarpit27 commented:
> Update the `Makefile`, as the commands are outdated (
> 
> https://github.com/0xPolygon/cdk/blob/94718685ad8a3df9d2975649b1c761aa53ed316b/test/Makefile#L7-L27
> 
> )

we dont need this now as we have moved to another script
removed in upcoming PR

Stefan-Ethernal reviewed (approved):
LGTM

temaniarpit27 committed:
update hash

rachit77 reviewed (approved):
💯

temaniarpit27 committed:
update hash
